src/Nethermind/Nethermind.Evm/Tracing/ITxTracer.cs

```diff
@@ -109,54 +256,172 @@ void ReportMemoryChange(UInt256 offset, in ReadOnlySpan<byte> data)
+        /// <param name="refundAddress"></param>
+        /// <remarks>Depends on <see cref="IsTracingActions"/></remarks>
         void ReportSelfDestruct(Address address, UInt256 balance, Address refundAddress);
 
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="gas"></param>
+        /// <param name="value"></param>
+        /// <param name="from"></param>
+        /// <param name="to"></param>
+        /// <param name="input"></param>
+        /// <param name="callType"></param>
+        /// <param name="isPrecompileCall"></param>
+        /// <remarks>Depends on <see cref="IsTracingActions"/></remarks>
         void ReportAction(long gas, UInt256 value, Address @from, Address to, ReadOnlyMemory<byte> input, ExecutionType callType, bool isPrecompileCall = false);
 
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="gas"></param>
+        /// <param name="output"></param>
+        /// <remarks>Depends on <see cref="IsTracingActions"/></remarks>
         void ReportActionEnd(long gas, ReadOnlyMemory<byte> output);
 
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="evmExceptionType"></param>
+        /// <param name="gasLeft"></param>
+        /// <remarks>Depends on <see cref="IsTracingActions"/></remarks>
         void ReportActionError(EvmExceptionType evmExceptionType, long gasLeft) => ReportActionError(evmExceptionType);
 
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="evmExceptionType"></param>
+        /// <remarks>Depends on <see cref="IsTracingActions"/></remarks>
         void ReportActionError(EvmExceptionType evmExceptionType);
 
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="gas"></param>
+        /// <param name="deploymentAddress"></param>
+        /// <param name="deployedCode"></param>
+        /// <remarks>Depends on <see cref="IsTracingActions"/></remarks>
         void ReportActionEnd(long gas, Address deploymentAddress, ReadOnlyMemory<byte> deployedCode);
 
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="blockHash"></param>
+        /// <remarks>Depends on <see cref="IsTracingBlockHash"/></remarks>
         void ReportBlockHash(Keccak blockHash);
 
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="byteCode"></param>
+        /// <remarks>Depends on <see cref="IsTracingCode"/></remarks>
         void ReportByteCode(byte[] byteCode);
 
         /// <summary>
         /// Special case for VM trace in Parity but we consider removing support for it
         /// </summary>
         /// <param name="refund"></param>
         /// <param name="gasAvailable"></param>
+        /// <remarks>Depends on <see cref="IsTracingInstructions"/></remarks>
         void ReportGasUpdateForVmTrace(long refund, long gasAvailable);
 
+        /// <summary>
+        ///
```

Comment:
Are these autogenerated?

Reply:
No but I decided to finally document it, yes it should be in separate PR.